### PR TITLE
Rename Storage Cluster to Storage Group. 

### PR DESCRIPTION
**Pull request by dvaldivia**
Rename Storage Cluster to Storage Group. Add Storage Cluster as the analogy to group multiple nodes.

Now we have the following resources:

- Tenant: A customer of m3 whose data may be spread across multiple `Storage Group`
- Storage Cluster: A representation of a group of nodes that translate to the analogy of real physical machines.
- Storage Group: A group of tenants packed together in a logical entity that lives inside a `Storage Cluster`. Multiple `Storage Groups` live inside a `Storage Cluster` in order to achieve maximum re-utilization of resources.
- Node: An individual machine that has a certain number of volumes and capacity

This is how data tables look now:

Screen Shot 2019-10-17 at 3 42 37 PM

### CODE REVIEW

**Comment by dvaldivia** [comment]
I realized I had a missing foreign key from `node_volumes` to `nodes`

Screen Shot 2019-10-17 at 4 49 28 PM

File: cluster/storage-cluster.go [hunk 1]
```diff
@@ -98,62 +104,64 @@ func ProvisionServicesForStorageCluster(storageCluster *StorageCluster) chan err
...
 
 	}()
 	return ch
 }
 
-// Returns a list of tenants that are allocated to the provided `StorageCluster`
-func GetListOfTenantsForSCluster(ctx *Context, sc *StorageCluster) chan []*StorageClusterTenant {
-	ch := make(chan []*StorageClusterTenant)
+// Returns a list of tenants that are allocated to the provided `StorageGroup`
+func GetListOfTenantsForStorageGroup(ctx *Context, sc *StorageGroup) chan []*StorageGroupTenant {
```
Comment (cesnietor) [inline cluster/storage-cluster.go]: can you also change `sc` like
`func GetListOfTenantsForStorageGroup(ctx *Context, sc *StorageGroup)`
to `func GetListOfTenantsForStorageGroup(ctx *Context, sg *StorageGroup)`
and all related usage

File: cluster/storage-cluster.go [hunk 2]
```diff
@@ -163,88 +171,88 @@ func GetListOfTenantsForSCluster(ctx *Context, sc *StorageCluster) chan []*Stora
...
+type StorageGroupTenantResult struct {
+	*StorageGroupTenant
 	Error error
 }
 
-// Creates a storage cluster in the DB
-func createTenantInStorageCluster(ctx *Context, tenant *Tenant, sc *StorageCluster) chan *StorageClusterTenantResult {
-	ch := make(chan *StorageClusterTenantResult)
+// Creates a storage group in the DB
+func createTenantInStorageGroup(ctx *Context, tenant *Tenant, sc *StorageGroup) chan *StorageGroupTenantResult {
```
Comment (cesnietor) [inline cluster/storage-cluster.go]: can you also change `sc` like
`func GetListOfTenantsForStorageGroup(ctx *Context, sc *StorageGroup)`
to `func GetListOfTenantsForStorageGroup(ctx *Context, sg *StorageGroup)`
and all related usage

File: cluster/tenants.go [hunk 3]
```diff
@@ -51,7 +53,7 @@ func AddTenant(name string, shortName string) error {
 		tx.Rollback()
 		return tenantResult.Error
 	}
-	fmt.Println(fmt.Sprintf("Registered as tenant %d\n", tenantResult.Tenant.ID))
+	fmt.Println(fmt.Sprintf("Registered as tenant %s\n", tenantResult.Tenant.ID.String()))
 
 	// find a cluster where to allocate the tenant
 	sc := <-SelectSCWithSpace(ctx)
```
Comment (cesnietor) [inline cluster/tenants.go]: should this also be changed ? all all related usages `sc`

File: cluster/cluster.go [hunk 4]
```diff
@@ -213,8 +214,8 @@ func CreateTenantSecrets(tenant *Tenant) error {
 	return nil
 }
 
-//CreateTenantService creates a service that will resolve to any of the hosts within the storage cluster this tenant lives in
-func CreateTenantService(sct *StorageClusterTenant) {
+//Creates a service that will resolve to any of the hosts within the storage group this tenant lives in
+func CreateTenantServiceInStorageGroup(sct *StorageGroupTenant) {
```
Comment (cesnietor) [inline cluster/cluster.go]: here `sct`

File: cluster/cluster.go [hunk 5]
```diff
@@ -250,8 +251,8 @@ func CreateTenantService(sct *StorageClusterTenant) {
 
 }
 
-func nodeNameForSCHostNum(sc *StorageCluster, hostNum string) string {
-	switch sc.ID {
+func nodeNameForSCHostNum(sg *StorageGroup, hostNum string) string {
```
Comment (cesnietor) [inline cluster/cluster.go]: `nodeNameForSCHostNum ` and usage

File: cluster/cluster.go [hunk 6]
```diff
@@ -287,48 +288,48 @@ const (
...
+func CreateDeploymentWithTenants(tenants []*StorageGroupTenant, sg *StorageGroup, hostNum string) error {
 	config := getConfig()
 	// creates the clientset to interact with kubernetes
 	clientset, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		return err
 	}
 
-	scHostName := fmt.Sprintf("sc-%d-host-%s", sc.ID, hostNum)
+	scHostName := fmt.Sprintf("sg-%d-host-%s", sg.Num, hostNum)
```
Comment (cesnietor) [inline cluster/cluster.go]: `scHostName ` to `sgHostName `

**Review by cesnietor — APPROVED** [review]

**Pull request merged**